server/pbenchinacan/load_keycloak.sh

```diff
@@ -1,110 +1,98 @@
+  -H "Authorization: Bearer ${ADMIN_TOKEN}" | jq -r '.value')
 
-echo "PBENCH_CLIENT_SECRET=$PBENCH_CLIENT_SECRET"
+echo "PBENCH_CLIENT_SECRET=${PBENCH_CLIENT_SECRET}"
 echo
 
-echo "Creating client role"
+echo "Creating an 'ADMIN' role under pbench-server client of the pbench realm"
 echo "--------------------"
 
-curl -i -X POST "http://$KEYCLOAK_HOST_PORT/admin/realms/pbench/clients/$CLIENT_ID/roles" \
-  -H "Authorization: Bearer $ADMIN_TOKEN" \
+curl -i -X POST "http://${KEYCLOAK_HOST_PORT}/admin/realms/pbench/clients/${CLIENT_ID}/roles" \
+  -H "Authorization: Bearer ${ADMIN_TOKEN}" \
   -H "Content-Type: application/json" \
   -d '{"name": "ADMIN"}'
 
-ROLE_ID=$(curl -s "http://$KEYCLOAK_HOST_PORT/admin/realms/pbench/clients/$CLIENT_ID/roles" \
-  -H "Authorization: Bearer $ADMIN_TOKEN" | jq -r '.[0].id')
+ROLE_ID=$(curl -s "http://${KEYCLOAK_HOST_PORT}/admin/realms/pbench/clients/${CLIENT_ID}/roles" \
+  -H "Authorization: Bearer ${ADMIN_TOKEN}" | jq -r '.[0].id')
 
-echo "ROLE_ID=$ROLE_ID"
+echo "ROLE_ID=${ROLE_ID}"
 echo
 
 echo "Creating user"
 echo "-------------"
 
-USER=$(curl -si -X POST "http://$KEYCLOAK_HOST_PORT/admin/realms/pbench/users" \
-  -H "Authorization: Bearer $ADMIN_TOKEN" \
+USER=$(curl -si -X POST "http://${KEYCLOAK_HOST_PORT}/admin/realms/pbench/users" \
+  -H "Authorization: Bearer ${ADMIN_TOKEN}" \
   -H "Content-Type: application/json" \
-  -d '{"username": "test_admin", "enabled": true, "credentials": [{"type": "password", "value": "123", "temporary": false}]}')
+  -d '{"username": "test_user", "enabled": true, "credentials": [{"type": "password", "value": "123", "temporary": false}]}')
 
-USER_ID=$(grep -o -e 'http://[^[:space:]]*' <<< $USER | sed -e 's|.*/||')
-echo "USER_ID=$USER_ID"
+USER_ID=$(grep -o -e 'http://[^[:space:]]*' <<< ${USER} | sed -e 's|.*/||')
+echo "USER_ID=${USER_ID}"
 echo
 
-echo "Setting client role to user"
+echo "Setting pbench-serve client role to the user created above"
 echo "---------------------------"
 
-curl -i -X POST "http://$KEYCLOAK_HOST_PORT/admin/realms/pbench/users/$USER_ID/role-mappings/clients/$CLIENT_ID" \
-  -H "Authorization: Bearer $ADMIN_TOKEN" \
+curl -i -X POST "http://${KEYCLOAK_HOST_PORT}/admin/realms/pbench/users/${USER_ID}/role-mappings/clients/${CLIENT_ID}" \
+  -H "Authorization: Bearer ${ADMIN_TOKEN}" \
   -H "Content-Type: application/json" \
-  -d '[{"id":"'"$ROLE_ID"'","name":"ADMIN"}]'
-
-echo "Getting user access token"
-echo "-------------------------"
-
-curl -s -X POST "http://$KEYCLOAK_HOST_PORT/realms/pbench/protocol/openid-connect/token" \
-  -H "Content-Type: application/x-www-form-urlencoded" \
-  -d "username=test_admin" \
-  -d "password=123" \
-  -d "grant_type=password" \
-  -d "client_secret=$PBENCH_CLIENT_SECRET" \
-  -d "client_id=pbench-client" | jq -r .access_token
-echo
-
-echo "---------"
-echo "PBENCH_CLIENT_SECRET=$PBENCH_CLIENT_SECRET"
-echo "---------"
+  -d '[{"id":"'"${ROLE_ID}"'","name":"ADMIN"}]'
```

Comment:
The double quotes immediately around `${ROLE_ID}` seem unnecessary and, in this context, excessive.